plugins/module_utils/network/sonic/config/interfaces/interfaces.py

```diff
@@ -282,26 +302,223 @@ def _state_deleted(self, want, have, diff):
+                                                                                   have,
+                                                                                   delete_all)
+        commands = []
+        if commands_del:
+            commands.extend(update_states(commands_del, "deleted"))
+        if commands_mer:
+            commands.extend(update_states(commands_mer, "merged"))
 
         return commands, requests
 
+    def handle_delete_interface_config(self, commands, have, delete_all=False):
+        requests = []
+        del_commands = []
+        mer_commands = []
+        if not commands:
+            return del_commands, mer_commands, requests
+
+        # Create URL and payload
+        for cmd in commands:
+            conf = cmd
+            name = conf['name']
+            have_conf = next((cfg for cfg in have if cfg['name'] == name), None)
+            if have_conf:
+                lp_key_set = set(conf.keys())
+                if name.startswith('Loopback'):
+                    if delete_all or len(lp_key_set) == 1:
+                        method = DELETE
+                        lpbk_url = url % quote(name, safe='')
+                        request = {"path": lpbk_url, "method": method}
+                        requests.append(request)
+
+                        del_commands.append({'name': name})
+
+                        continue
+
+                if len(lp_key_set) == 1:
+                    conf = have_conf
+
+                request = self.build_delete_description_request(conf, have_conf)
+                if request:
+                    requests.append(request)
+                    mer_commands.append(conf)
+
+                request = self.build_delete_enabled_request(conf, have_conf)
+                if request:
+                    requests.append(request)
+                    mer_commands.append(conf)
+
+                request = self.build_delete_mtu_request(conf, have_conf)
+                if request:
+                    requests.append(request)
+                    mer_commands.append(conf)
+
+                request = self.build_delete_fec_request(conf, have_conf)
+                if request:
+                    requests.append(request)
+                    mer_commands.append(conf)
+
+                request = self.build_delete_speed_request(conf, have_conf)
+                if request:
+                    requests.append(request)
+                    mer_commands.append(conf)
+
+                request = self.build_delete_autoneg_request(conf, have_conf)
+                if request:
+                    requests.append(request)
+                    mer_commands.append(conf)
+
+                request = self.build_delete_advertised_speed_request(conf, have_conf)
+                if request:
+                    requests.append(request)
+                    mer_commands.append(conf)
+
+        return del_commands, mer_commands, requests
```

Comment:
For non-loopback interfaces, del_commands will be empty.
mer_commands will be there.
So _state_deleted will have only "merged" in commands.

Reply:
Uploaded regression test result pdf file.

Reply:
Please see the above replay comment. Thanks.